deployment-templates/compute-engine/client/forseti-instance-client.py

```diff
@@ -24,9 +49,30 @@ def GenerateConfig(context):
         "git clone {src_path}.git".format(
             src_path=context.properties['src-path']))
 
-    FORSETI_VERSION = (
-        "git checkout {forseti_version}".format(
-            forseti_version=context.properties['forseti-version']))
+    patch_search_expression = get_patch_search_expression(context.properties['forseti-version'])
+    if patch_search_expression:
+        CHECKOUT_FORSETI_VERSION = (
+            """versions=$(git tag -l {patch_search_expression})
+versions=(${{versions//;/ }})
+for version in "${{versions[@]}}"
+do
+segments=(${{version//./ }})
+patch=${{segments[2]}}
+patch=${{patch: 0: 2}}
+patch=$(echo $patch | sed 's/[^0-9]*//g')
+#  latest_version is an array [full_version, patch_number]
+if !((${{#latest_version[@]}})) || ((patch > ${{latest_version[1]}}));
+then
+  latest_version=($version $patch)
+fi
+done
+git checkout ${{latest_version[0]}}"""
+
+                .format(patch_search_expression=patch_search_expression))
```

Comment:
Do you need to append this `.format()`directly to the end of the bash string?

Reply:
no it works